core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -73,11 +76,12 @@ private static byte[] prependEip1559TransactionType(byte[] bytesToSign) {
         .array();
   }
 
-  private SignatureData sign(final byte[] bytesToSign) {
-    final SecpArtifactSignature artifactSignature =
-        (SecpArtifactSignature) signer.sign(Bytes.of(bytesToSign));
+  private SignatureData sign(final String eth1Address, final byte[] bytesToSign) {
+    final String bytesSigned =
+        secpSigner.sign(normaliseIdentifier(eth1Address), Bytes.of(bytesToSign)).orElseThrow();
```

Comment:
We should throw JsonRpcException(SIGNING_FROM_IS_NOT_AN_UNLOCKED_ACCOUNT) so that the appropriate error is returned for the JsonRpc response

Reply:
Good catch